authentication_sasl.go

```diff
@@ -58,3 +59,27 @@ func (src *AuthenticationSASL) Encode(dst []byte) []byte {
 
 	return dst
 }
+
+// MarshalJSON implements encoding/json.Marshaler.
+func (src AuthenticationSASL) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Type           string
+		AuthMechanisms []string
+	}{
+		Type:           "AuthenticationSASL",
+		AuthMechanisms: src.AuthMechanisms,
+	})
+}
+
+// UnmarshalJSON implements encoding/json.Unmarshaler.
+func (dst *AuthenticationSASL) UnmarshalJSON(data []byte) error {
```

Comment:
Why did you implement this function? Isn't it going to do the exact same if it fallbacks to Go's default unmarshaling given the Type is never used?

Reply:
With the implementation:

```
B{"Type":"AuthenticationSASL","AuthMechanisms":["SCRAM-SHA-256"]}
```

Without:

```
B{"AuthMechanisms":["SCRAM-SHA-256"]}
```

Reply:
thanks, will delete.